.github/scripts/veristat-compare.py

```diff
@@ -38,165 +38,212 @@
 import io
 import os
 import sys
+import re
 import csv
 import logging
 import argparse
-from functools import reduce
+import enum
 from dataclasses import dataclass
+from typing import Final, List
+
+
+TRESHOLD_PCT: Final[int] = 0
+
+SUMMARY_HEADERS = ["File", "Program", "Verdict", "States Diff (%)"]
+
+# expected format: +0 (+0.00%) / -0 (-0.00%)
+TOTAL_STATES_DIFF_REGEX = \
+    r"(?P<absolute_diff>[+-]\d+) \((?P<percentage_diff>[+-]\d+\.\d+)\%\)"
+
+
+TEXT_SUMMARY_TEMPLATE: Final[str] = """
+# {title}
+
+{table}
+""".strip()
+
+HTML_SUMMARY_TEMPLATE: Final[str] = """
+# {title}
+
+<details>
+<summary>Click to expand</summary>
+
+{table}
+</details>
+""".strip()
+
+
+class VeristatFields(str, enum.Enum):
+    FILE_NAME = "file_name"
+    PROG_NAME = "prog_name"
+    VERDICT_OLD = "verdict_base"
+    VERDICT_NEW = "verdict_comp"
+    VERDICT_DIFF = "verdict_diff"
+    TOTAL_STATES_OLD = "total_states_base"
+    TOTAL_STATES_NEW = "total_states_comp"
+    TOTAL_STATES_DIFF = "total_states_diff"
+
+    @classmethod
+    def headers(cls) -> List[str]:
+        return [
+            cls.FILE_NAME,
+            cls.PROG_NAME,
+            cls.VERDICT_OLD,
+            cls.VERDICT_NEW,
+            cls.VERDICT_DIFF,
+            cls.TOTAL_STATES_OLD,
+            cls.TOTAL_STATES_NEW,
+            cls.TOTAL_STATES_DIFF,
+        ]
 
-TRESHOLD_PCT = 0
-
-HEADERS = ['file_name', 'prog_name', 'verdict_base', 'verdict_comp',
-           'verdict_diff', 'total_states_base', 'total_states_comp',
-           'total_states_diff']
-
-FILE        = 0
-PROG        = 1
-VERDICT_OLD = 2
-VERDICT_NEW = 3
-STATES_OLD  = 5
-STATES_NEW  = 6
-
-# Given a table row, compute relative increase in the number of
-# processed states.
-def compute_diff(v):
-    old = int(v[STATES_OLD]) if v[STATES_OLD] != 'N/A' else 0
-    new = int(v[STATES_NEW]) if v[STATES_NEW] != 'N/A' else 0
-    if old == 0:
-        return 1
-    return (new - old) / old
 
 @dataclass
 class VeristatInfo:
     table: list
     changes: bool
     new_failures: bool
 
-# Read CSV table expecting the above described format.
-# Return VeristatInfo instance.
-def parse_table(csv_filename):
+    def get_results_title(self) -> str:
+        if self.new_failures:
+            return "There are new veristat failures"
+
+        if self.changes:
+            return "There are changes in verification performance"
+
+        return "No changes in verification performance"
+
+    def get_results_summary(self, html: bool = False) -> str:
+        title = self.get_results_title()
+        if not self.table:
+            return f"# {title}\n"
```

Comment:
as a follow-up, it would be good to report failure/warning with color codes so it is made more obvious in GH.

Reply:
Github markup doesn't support colors: https://github.com/github/markup/issues/1440

Reply:
The text content can be "colorized" by mean of special tags, see https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-a-warning-message

The markup one can use simple to signal success/failure: https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/

Reply:
It's not exactly "colors", but I got what you mean. I will cover it in a separate PR.